src/nodes/QskTickmarksMetrics.h

```diff
@@ -0,0 +1,131 @@
+/******************************************************************************
+ * QSkinny - Copyright (C) 2016 Uwe Rathmann
+ *           SPDX-License-Identifier: BSD-3-Clause
+ *****************************************************************************/
+
+#ifndef QSK_TICKMARKS_METRICS_H
+#define QSK_TICKMARKS_METRICS_H
+
+#include "QskFunctions.h"
+
+#include <qmetatype.h>
+
+class QSK_EXPORT QskTickmarksMetrics
+{
+    Q_GADGET
+    Q_PROPERTY( qreal majorRatio READ majorRatio WRITE setMajorRatio )
+    Q_PROPERTY( qreal mediumRatio READ mediumRatio WRITE setMediumRatio )
+    Q_PROPERTY( qreal minorRatio READ minorRatio WRITE setMinorRatio )
+
+  public:
+    constexpr QskTickmarksMetrics() noexcept = default;
+    constexpr QskTickmarksMetrics( qreal minorRatio, qreal mediumRatio, qreal majorRatio ) noexcept;
+    constexpr QskTickmarksMetrics( const QskTickmarksMetrics& ) noexcept = default;
+    constexpr QskTickmarksMetrics( QskTickmarksMetrics&& ) noexcept = default;
+
+    constexpr QskTickmarksMetrics& operator=( const QskTickmarksMetrics& ) noexcept = default;
+    constexpr QskTickmarksMetrics& operator=( QskTickmarksMetrics&& ) noexcept = default;
+
+    constexpr Q_REQUIRED_RESULT bool operator==( const QskTickmarksMetrics& rhs ) const noexcept;
+    constexpr Q_REQUIRED_RESULT bool operator!=( const QskTickmarksMetrics& rhs ) const noexcept;
+
+    constexpr Q_REQUIRED_RESULT qreal majorRatio() const noexcept;
+    constexpr Q_REQUIRED_RESULT qreal mediumRatio() const noexcept;
+    constexpr Q_REQUIRED_RESULT qreal minorRatio() const noexcept;
+
+    constexpr void setMajorRatio( qreal ratio ) noexcept;
+    constexpr void setMediumRatio( qreal ratio ) noexcept;
+    constexpr void setMinorRatio( qreal ratio ) noexcept;
+
+    constexpr Q_REQUIRED_RESULT qreal operator[]( int index ) const noexcept;
+    constexpr Q_REQUIRED_RESULT qreal& operator[]( int index ) noexcept;
+
+  private:
+    qreal m_minorRatio = 0.7;
+    qreal m_mediumRatio = 0.85;
+    qreal m_majorRatio = 1.0;
+};
```

Comment:
@uwerat I was thinking about adding `Qt::SizeMode` but this would need more adaptions in the `QskTickmarksNode`. What do you think?

Reply:
Guess setting the length of the ticks as absolute values ( pixels/dp ) is what a user usually want to do.

IIRC the current implementation relates the tick lengths to a metric, that is given as a skin hint. IMHO this does not make much sense anymore as QskTickmarksMetrics allows to pass the tick lengths as skin hint instead.